tests/hooks/event-conversion-utils.spec.ts

```diff
@@ -707,9 +803,426 @@ describe("formatDates", () => {
+      },
+      {
+        endDate: "",
+        institution: "Kihn - Rolfson",
+        location: "YouTube",
+        speaker: "Deondre Halvorson",
+        title: "Ea commodi earum id.",
+        to: "http://jam-packed-unity.com",
+        abstract: "Qui facere assumenda ducimus.",
+        date: "June 9, 2023",
+        image: "/images/events/downloaded/undefined.jpg",
+        startDate: "2023-06-09",
+      },
+    ]);
+  });
+
+  test("fetch in date range events descending", async () => {
+    const eventsAirtableRecords = new FakeEventsAirtableRecords(
+      "testApiKey",
+      "testView",
+      undefined,
+      mockRecordFields
+    );
+
+    const getEventsQueryMock = vi.fn().mockReturnValue({
+      eachPage: async (
+        callBack: (records: any, nextPage: any) => Promise<void>
+      ) => {
+        await callBack(fakeRecords, () => {});
+
+        return Promise.resolve();
+      },
+    });
+
+    eventsAirtableRecords.getEventsQuery = getEventsQueryMock;
+    eventsAirtableRecords.isEventInDateRange = vi.fn().mockReturnValue(true);
+
+    (axios.get as any).mockResolvedValue({
+      data: "Omnis non molestiae voluptatem beatae soluta nobis ducimus.",
+    });
+    const result = await eventsAirtableRecords.fetchSeminarSeriesEvents(-10);
+
+    expect(result).toStrictEqual([
+      {
+        endDate: "",
+        institution: "Kihn - Rolfson",
+        location: "YouTube",
+        speaker: "Deondre Halvorson",
+        title: "Ea commodi earum id.",
+        to: "http://jam-packed-unity.com",
+        abstract: "Qui facere assumenda ducimus.",
+        date: "June 9, 2023",
+        image: "/images/events/downloaded/undefined.jpg",
+        startDate: "2023-06-09",
+      },
+      {
+        endDate: "",
+        institution: "Okuneva Group",
+        location: "YouTube",
+        speaker: "Lela Bahringer",
+        title: "delectus eaque consectetur",
+        to: "http://overlooked-sleeping.org",
+        abstract: "",
+        date: "July 29, 2022",
+        image: "/images/events/downloaded/undefined.jpg",
+        startDate: "2022-07-29",
+      },
+    ]);
+  });
+
+  test("If no recordFields should get from getAllFieldNames", async () => {
+    const eventsAirtableRecords = new FakeEventsAirtableRecords(
+      "testApiKey",
+      "testView"
+    );
+
+    const getEventsQueryMock = vi.fn().mockReturnValue({
+      eachPage: (callBack: (records: any, nextPage: any) => void) => {
+        callBack(fakeRecords, () => {});
+
+        return Promise.resolve();
+      },
+    });
+
+    const getAllFieldNamesMock = vi.fn().mockReturnValue(mockRecordFields);
+
+    eventsAirtableRecords.getEventsQuery = getEventsQueryMock;
+    eventsAirtableRecords.getAllFieldNames = getAllFieldNamesMock;
+
+    await eventsAirtableRecords.fetchSeminarSeriesEvents(10);
+
+    expect(eventsAirtableRecords.recordFields).toStrictEqual(mockRecordFields);
+  });
+
+  test("test fallback cases", async () => {
```

Comment:
I think that we should explain what specific fallback cases we're testing here.

Reply:
I changed the description